dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java

```diff
@@ -118,6 +122,38 @@ public void testAddDataValueWithBody()
         assertStatus( HttpStatus.CREATED, response );
     }
 
+    /**
+     * Check if the dataValueSet endpoint return correct fileName.
+     */
+    @Test
+    void testGetDataValueSetJsonWithAttachment()
+    {
+        String dsId = assertStatus( HttpStatus.CREATED,
+            POST( "/dataSets/",
+                "{'name':'My data set', 'periodType':'Monthly', 'dataSetElements':[{'dataElement':{'id':'"
+                    + dataElementId + "'}}]}" ) );
+
+        String body = String.format( "{" +
+            "'dataElement':'%s'," +
+            "'categoryOptionCombo':'%s'," +
+            "'period':'20220102'," +
+            "'orgUnit':'%s'," +
+            "'value':'24'," +
+            "'comment':'OK'}",
+            dataElementId, categoryOptionComboId, orgUnitId );
+
+        HttpResponse response = POST( "/dataValues", body );
+        assertStatus( HttpStatus.CREATED, response );
+        switchToUserWithOrgUnitDataView( "testUser", orgUnitId );
+        String url = "/dataValueSets?orgUnit=" + orgUnitId + "&startDate=2022-01-01&endDate=2022-01-30&dataSet=" + dsId
+            +
+            "&format=json&compression=zip&attachment=dataValues.json.zip";
+        MvcResult dataValueResponse = webRequestWithMvcResult(
+            buildMockRequest( HttpMethod.GET, url, Collections.emptyList(), null, null ) );
+        assertTrue( dataValueResponse.getResponse().getHeader( "Content-Disposition" )
```

Comment:
Wonder if you added the `webRequestWithMvcResult` so you could read the header? In that case this is not needed. You can read headers from the `HttpResonse` returned by the `GET` method.